### PR TITLE
Add docker-image workflow

### PR DESCRIPTION
**Pull request by symeneses**
**What is this PR about?**
It solves #5939 

It add a workflow `docker-image`, The workflow get the labels and tags, build the image, test it and push it to Docker Hub. The workflow will be triggered when there are new releases.

It's needed to have the secrets with the Docker Hub user `DOCKERHUB_USER` and token `DOCKERHUB_TOKEN`.

The workflow was tested using [act](https://github.com/nektos/act). The test image is in a [test image](https://hub.docker.com/repository/docker/symeneses/test) where you can see the tags (latest and version) created. These tags can be modified if needed.

#DataUmbrellaPyMCSprint

...

**Checklist**
+ [ x] Explain important implementation details 👆
+ [x ] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [ x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [ x] Are the changes covered by tests and docstrings?
+ [ x] Fill out the short summary sections 👇

### CODE REVIEW

**Comment by michaelosthege** [comment]
To test this, we should (temporarily) change the trigger condition to
```yaml
on:
  push:
```
...and turn off actually pushing it.
I believe this is covered by the example here? https://github.com/docker/build-push-action#git-context

They also write that the checkout action is not needed, because the `docker/build-and-push` action does a git checkout automatically?

In the meantime I'll try to figure out who knows our `DOCKERHUB_USER` and `DOCKERHUB_TOKEN`, because they are not yet set as repository secrets.

**Comment by michaelosthege** [comment]
> The workflow was tested using [act](https://github.com/nektos/act). The test image is in a [test image](https://hub.docker.com/repository/docker/symeneses/test) where you can see the tags (latest and version) created. These tags can be modified if needed.

The image appears to be private? It's asking me to log in..

**Comment by twiecki** [comment]
> In the meantime I'll try to figure out who knows our DOCKERHUB_USER and DOCKERHUB_TOKEN, because they are not yet set as repository secrets.

I own the docker organization and now created and added these secrets, so this should work now. Can we test it somehow or should we just merge?

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: scripts/Dockerfile
+          load: true
+          tags: ${{ env.CONTAINER_NAME }}
+
+      - name: Test
+        run: |
+           docker run --rm ${{ env.CONTAINER_NAME }} bash
```
Comment (ferrine) [inline .github/workflows/docker-image.yml]: Try importing pymc

**Comment by symeneses** [comment]
> They also write that the checkout action is not needed, because the `docker/build-and-push` action does a git checkout automatically?

I tested it without the checkout but it failed.

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (symeneses) [reply .github/workflows/docker-image.yml]: Done ✅

**Comment by symeneses** [comment]
> > The workflow was tested using [act](https://github.com/nektos/act). The test image is in a [test image](https://hub.docker.com/repository/docker/symeneses/test) where you can see the tags (latest and version) created. These tags can be modified if needed.
> 
> The image appears to be private? It's asking me to log in..

You should be able to see login in with any user. The image looks like this:
Screenshot 2022-07-11 at 11 26 18

**Comment by symeneses** [comment]
> > In the meantime I'll try to figure out who knows our DOCKERHUB_USER and DOCKERHUB_TOKEN, because they are not yet set as repository secrets.
> 
> I own the docker organization and now created and added these secrets, so this should work now. Can we test it somehow or should we just merge?

It can be tested using [act](https://github.com/nektos/act) executing this command:
```bash
act release -j build-container -v
```

For that I changed the image name `name=pymc/pymc` to `name=symeneses/test`. Also, as my laptop is M1, while testing I had to add `platforms: linux/x86_64` in the  steps `Build and load image` and `Build and push`.

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
I'm merging this, we can always keep working on it in follow-up PRs. But will this now trigger a build we can test or do we need to make a release for that?

**Comment by symeneses** [comment]
> I'm merging this, we can always keep working on it in follow-up PRs. But will this now trigger a build we can test or do we need to make a release for that?

It will be triggered only by release

**Comment by twiecki** [comment]
Just cut a new release, let's see!

**Comment by twiecki** [comment]
Does not look promising https://github.com/pymc-devs/pymc/actions/runs/2676293391

**Pull request merged**